ui/app/src/model/project-client.ts

```diff
@@ -112,3 +112,11 @@ export function useDeleteProjectMutation() {
     },
   });
 }
+
+export function fetchProject(name: string) {
```

Comment:
It is preferrable to make a useProject function using tanstack as it is the direction taken by all the clients. 

If you want to receive a promise, you should use mutation and mutateAsync method

Reply:
https://tanstack.com/query/v3/docs/react/guides/mutations
https://tanstack.com/query/v3/docs/react/guides/mutations#promises

Reply:
mutate or mutateAsync is not the correct way when you want to fetch data :/. It's more used when you update or create data

Reply:
I don't really see the value of doing that. Using fetch directly is doing exactly what is expected and it is simpler. Yes it doesn't have a cache or whatever react-query has, but it's not necessary here. The simpler the better

Reply:
Yes we need to discuss about that custom cache in datasources. I think it should be migrated to react-query. 
I don´t understand what justified the creation of this cache wrapper. Is this historical ? because react-query was not used before? Or is there a specific feature that you make and which is not supported by react-query? Maybe you can explain me. Steven told me he tried also to migrate it to react-query some day, without success.

Reply:
I guess we can keep it like that as it seems you don't want to have cache for project guard...